src/components/VmsList/Vms.js

```diff
@@ -1,25 +1,69 @@
 import React from 'react'
+import ReactDOM from 'react-dom'
 import PropTypes from 'prop-types'
+import { Link } from 'react-router-dom'
+import { Icon, Button } from 'patternfly-react'
 
 import { connect } from 'react-redux'
 
 import style from './style.css'
+import sharedStyle from '_/components/sharedStyle.css'
 import Vm from './Vm'
 import Pool from './Pool'
 import ScrollPositionHistory from '../ScrollPositionHistory'
 import { getByPage } from '_/actions'
 import { filterVms, sortFunction } from '_/utils'
+import { msg } from '_/intl'
 import InfiniteScroll from 'react-infinite-scroller'
 import Loader, { SIZES } from '../Loader'
 
+const VmsSettingsButton = ({ checkedVms = [] }) => {
+  const container = document.getElementById('vm-settings-btn-box')
+  if (container) {
+    return ReactDOM.createPortal(
```

Comment:
This is being portaled to the toolbar because you're tracking the selection state in the Vms component itself right?  If you kept track of the VM selection in redux state, the action button wouldn't need to do the portaling and be tied to the Vms component so tightly.

Reply:
I would like to avoid redux state in this case, because checking VMs is very closely related to component and using URL for me is more useful. But PR still not support updating URL after selecting VM on settings page. I'll do this in next update.